src/node_options.cc

```diff
@@ -1233,6 +1233,13 @@ void GetEmbedderOptions(const FunctionCallbackInfo<Value>& args) {
   Local<Context> context = env->context();
   Local<Object> ret = Object::New(isolate);
 
+  if (ret->Set(context,
+           FIXED_ONE_BYTE_STRING(
+            env->isolate(), "hasPrepareStackTraceCallback"),
+           Boolean::New(isolate,
+            !env->prepare_stack_trace_callback().IsEmpty()))
```

Comment:
This is overridden when the realm is bootstrapped though, unless `SetIsolateUpForNode()` is somehow called after `CreateEnvironment()` this would not have worked?

https://github.com/nodejs/node/blob/fe219e043827f5885d9c764a95324ded14ea94bf/lib/internal/bootstrap/realm.js#L449

I wonder if this flag should've been moved into `EnvironmentFlags` instead (and if that's too breaking, make `SHOULD_NOT_SET_PREPARE_STACK_TRACE_CALLBACK` a dummy flag there), and the `GetEmbedderOptions()` could query that flag stored into the `Environment` instead. At bootstrap time, we could just skip `setPrepareStackTraceCallback` if that flag is true.

Reply:
@joyeecheung ah, we do call `CreateEnvironment()` before `SetIsolateUpForNode()` - I can switch that around.